tools/src/main/java/com/nedap/archie/creation/ExampleJsonInstanceGenerator.java

```diff
@@ -0,0 +1,495 @@
+        for (BmmProperty property : properties.values()) {
+            if (property.getMandatory() && !result.containsKey(property.getName())) {
+                addRequiredProperty(result, property);
+            }
+        }
+    }
+
+    private void addRequiredProperty(Map<String, Object> result, BmmProperty property) {
+        BmmType type = property.getType();
+        BmmClass propertyClass = type.getBaseClass();
+        if (property instanceof BmmContainerProperty) {
+            List<Map<String, Object>> children = new ArrayList<>();
+            result.put(property.getName(), children);
+        } else if (propertyClass instanceof BmmEnumerationInteger) {
+            result.put(property.getName(), 0);
+        } else if (propertyClass instanceof BmmEnumerationString) {
+            result.put(property.getName(), "string");
+        } else {
+            String actualType = type.getTypeName();
+            BmmClass classDefinition1 = bmm.getClassDefinition(BmmDefinitions.typeNameToClassKey(actualType));
+            if(classDefinition1 != null && classDefinition1.isPrimitiveType()) {
+                if (aomProfile.getRmPrimitiveTypeEquivalences().get(type.getTypeName()) != null) {
+                    actualType = aomProfile.getRmPrimitiveTypeEquivalences().get(type.getTypeName());
+                }
+                result.put(property.getName(), generatePrimitiveTypeExample(actualType));
+            } else {
+                result.put(property.getName(), constructExampleType(actualType));
+            }
+        }
+    }
+
+    private Map<String, Object> constructExampleType(String actualType) {
+        Map<String, Object> custom = generateCustomExampleType(actualType);
+        if(custom != null) {
+            return custom;
+        }
+        Map<String, Object> result = new LinkedHashMap<>();
+        String className = getConcreteTypeName(actualType);
+        BmmClass classDefinition = bmm.getClassDefinition(BmmDefinitions.typeNameToClassKey(actualType));
+        result.put("@type", className);
+        if(classDefinition != null) {
+            addRequiredPropertiesFromBmm(result, classDefinition);
+        }
+        return result;
+    }
+
+    private Object generatePrimitiveTypeExample(String typeName) {
+        switch(typeName.toLowerCase()) {
+            case "string":
+                return "string";
+            case "real":
+                return 42.0d;
+            case "integer":
+                return 42;
+            case "date":
+                return "2018-01-01";
+            case "date_time":
+                return "2018-01-01T12:00:00+00:00";
+            case "time":
+                return "12:00:00";
+            case "duration":
+                return "PT10m";
+            case "boolean":
+                return "true";
+        }
+        return "unknown primitive type " + typeName;
+    }
+
+    private void generateCPrimitive(List<Object> children, CPrimitiveObject child) {
```

Comment:
Better if this would just generate and return the primitive object instead of also adding it to the children list, more in line with the other generate methods and the name of the method

Reply:
fixed